### PR TITLE
Expressions on facets

### PR DESCRIPTION
**Pull request by jorgensd**
Makes it possible to use expression tabulation on facets.

As for exterior facet integrals, they are determined by the local facet index of a given cell.

For an Expression to be considered a facet expression, the evaluation points has to be of the topological dimension of the facet

### CODE REVIEW

File: ffcx/codegeneration/expression_generator.py [hunk 1]
```diff
@@ -64,14 +64,19 @@ def generate_geometry_tables(self):
         # Currently we only support circumradius
         ufl_geometry = {
             ufl.geometry.ReferenceCellVolume: "reference_cell_volume",
+            ufl.geometry.ReferenceNormal: "reference_facet_normals",
         }
+        # Types that are only supported for facet expressions
```
Comment (michalhabera) [inline ffcx/codegeneration/expression_generator.py:69]: What is this comment related to?

File: ffcx/ir/elementtables.py [hunk 2]
```diff
@@ -93,7 +93,10 @@ def get_ffcx_table_values(
 
     if integral_type == "expression":
         # FFCx tables for expression are generated as interior cell points
```
Comment (michalhabera) [inline ffcx/ir/elementtables.py:95]: This comment needs updating.

**Review by michalhabera — CHANGES_REQUESTED** [review]
This will be very useful!

File: ffcx/codegeneration/ufcx.h [hunk 3]
```diff
@@ -361,12 +361,11 @@ extern "C"
     /// Number of evaluation points
     int num_points;
 
-    /// Dimension of evaluation point, i.e. topological dimension of
-    /// reference cell
-    int topological_dimension;
+    /// Dimension of evaluation point
+    int point_dimension;
```
[the remark this answers is not in the document]
Comment (jorgensd) [reply ffcx/codegeneration/ufcx.h:365]: Can call it entity dimension. Topological dimension is just a bit misleading as it is unclear what we are referring to.

File: ffcx/codegeneration/expression_generator.py (hunk 1 above)
Comment (jorgensd) [reply ffcx/codegeneration/expression_generator.py:69]: Leftover from a previous iteration of this code.

**Review by IgorBaratta — APPROVED** [review]

**Review by michalhabera — APPROVED** [review]

File: .github/workflows/dolfinx-tests.yml [hunk 4]
```diff
@@ -51,7 +51,7 @@ jobs:
         with:
           path: ./dolfinx
           repository: FEniCS/dolfinx
-          ref: main
+          ref: dokken/facet-expressions
```
Comment (jorgensd) [inline .github/workflows/dolfinx-tests.yml:54]:
```suggestion
          ref: main
```
Revert branch to main prior to merge
